File: src/main/java/com/mrozowski/elasticsearchdemo/domain/port/ProductRepository.java
package com.mrozowski.elasticsearchdemo.domain.port;

import com.mrozowski.elasticsearchdemo.domain.model.Product;
import org.springframework.data.domain.Page;

import java.util.List;
import java.util.Optional;

public interface ProductRepository {

  Optional<Product> findById(String id);

  String addProduct(Product product);

  List<Product> searchByNameOrDescription(String value);

  List<String> suggest(String fraze);
}

File: src/main/java/com/mrozowski/elasticsearchdemo/domain/port/ProductRepository.kt
package com.mrozowski.elasticsearchdemo.domain.port

import com.mrozowski.elasticsearchdemo.domain.model.Product
import com.mrozowski.elasticsearchdemo.domain.model.SearchCommand
import org.springframework.data.domain.Page

interface ProductRepository {
    fun findById(id: String): Product?
    fun addProduct(product: Product): String
    fun searchByNameOrDescription(value: String): List<Product>
    fun search(command: SearchCommand): Page<Product>
    fun suggest(fraze: String): Set<String>
    fun addProducts(products: List<Product>)
}
